ddtrace/tracer/tracer.go

```diff
@@ -308,7 +311,9 @@ func (t *tracer) flushTraces() {
 
 // flushErrors will process log messages that were queued
 func (t *tracer) flushErrors() {
-	logErrors(t.errorBuffer)
+	if t.errorReporter != nil {
+		t.errorReporter(t.errorBuffer)
```

Comment:
I think this should go [here](https://github.com/DataDog/dd-trace-go/blob/d50d57eca7aafda57e6c39fb8901525cda1ebba0/ddtrace/tracer/errors.go#L67). I wouldn't want people ending up having the tracer blocking their application due to not handling the channel properly. I'd like us (inside this package) to continue handling input from the channel.

Reply:
Either is fine:
- the `errorReporter` is pushed each error as it happens
- the `errorReporter` is pushed an aggregation of the errors after a quantum

The spirit of this is to allow us to report the errors ourselves, conceivably the aggregated error is basically the same as having N individual errors, but it would still be nice to have the individual errors.

Reply:
Let's do the latter. I think the current version doesn't serve for an intuitive API. Mainly because when you receive a channel from any API you'd expect it to be actively receiving whatever its type is (in this case `error`), which is not at all what happens here. In our case, the function is only called occasionally with a channel having 0 or more buffered entries. This might be a source of confusion to some.

Let's just add a second parameter to `logErrors` which will be of type `func(error)`. This function will then be called with whatever errors occur. If we want to improve or customise the way we aggregate errors we can work on that later on.

What do you think?